src/estimation/first_step_estimation/est_mortality.py

```diff
@@ -135,183 +132,70 @@ def estimate_mortality(paths_dict, specs):
     age_coef = params.loc["age", "value"]
-    health1_edu1_coef = params.loc["health1_edu1", "value"]
-    health1_edu0_coef = params.loc["health1_edu0", "value"]
-    health0_edu1_coef = params.loc["health0_edu1", "value"]
-    health0_edu0_coef = params.loc["health0_edu0", "value"]
+    coefficients = params["value"]
 
+    # Compute lambda using health factors
     lambda_ = np.exp(
-        cons
-        + health1_edu1_coef * health1_edu1
-        + health1_edu0_coef * health1_edu0
-        + health0_edu1_coef * health0_edu1
-        + health0_edu0_coef * health0_edu0
+        cons + sum(coefficients[f"health{i}_edu{j}"] * factor for i, j, factor in zip([1, 1, 0, 0], [1, 0, 1, 0], health_factors))
     )
     age_contrib = np.exp(age_coef * age) - 1
 
     return -(lambda_ * age_contrib) / age_coef
 
-
 def loglike(params, data):
+    """
+    Log-likelihood calculation.
+    """
     start_age = data["start_age"]
     age = data["age"]
     event = data["event_death"]
-    death = data["event_death"].astype(bool)
-    health1_edu1 = data["health1_edu1"]
-    health1_edu0 = data["health1_edu0"]
-    health0_edu1 = data["health0_edu1"]
-    health0_edu0 = data["health0_edu0"]
-    start_health0_edu0 = data["start_health0_edu0"]
-    start_health0_edu1 = data["start_health0_edu1"]
-    start_health1_edu0 = data["start_health1_edu0"]
-    start_health1_edu1 = data["start_health1_edu1"]
+    death = event.astype(bool)
 
-    # initialize contributions as an array of zeros
-    contributions = np.zeros_like(age)
+    # Extract health factors
+    health_factors = [data[f"health{i}_edu{j}"] for i, j in zip([1, 1, 0, 0], [1, 0, 1, 0])]
+    start_health_factors = [data[f"start_health{i}_edu{j}"] for i, j in zip([1, 1, 0, 0], [1, 0, 1, 0])]
 
-    # calculate contributions
-    contributions[death] = log_density_function(
-        age[death],
-        health1_edu1[death],
-        health1_edu0[death],
-        health0_edu1[death],
-        health0_edu0[death],
-        params,
-    )
-    contributions[~death] = log_survival_function(
-        age[~death],
-        health1_edu1[~death],
-        health1_edu0[~death],
-        health0_edu1[~death],
-        health0_edu0[~death],
-        params,
-    )
-    contributions -= log_survival_function(
-        start_age,
-        start_health1_edu1,
-        start_health1_edu0,
-        start_health0_edu1,
-        start_health0_edu0,
-        params,
-    )
+    # Initialize contributions as an array of zeros
+    contributions = np.zeros_like(age)
 
-    # print the death and not death contributions
-    print(
-        "Iteration:",
-        globals()["Iteration"],
-        "Total contributions:",
-        contributions.sum(),
-    )
+    # Calculate contributions
+    contributions[death] = log_density_function(age[death], [f[death] for f in health_factors], params)
+    contributions[~death] = log_survival_function(age[~death], [f[~death] for f in health_factors], params)
```

Comment:
These are only four health factors? Maybe you can also do just four contributions without for loop?